cpu/native/periph/rtc.c

```diff
@@ -167,9 +167,14 @@ int rtc_get_time(struct tm *ttime)
     }
 
     _native_syscall_enter();
-    t = time(NULL) + _native_rtc_offset;
+    gettimeofday(&tv, NULL);
```

Comment:
would `clock_gettime` be a better fit?

Reply:
Why do you think so?

Reply:
we would be less dependent  on whats happening on the system around RIOT if we choose `CLOCK_MONOTONIC` less of the linux smarts would leak into RIOT native

Reply:
> On Linux, that  point corresponds  to  the number of seconds that the system has been running since it was booted.

Hm, I think I'd want the RTC to be the actual wall clock, not seconds since boot (and end up somewhere in 1970).

Reply:
> > On Linux, that  point corresponds  to  the number of seconds that the system has been running since it was booted.
> 
> Hm, I think I'd want the RTC to be the actual wall clock, not seconds since boot (and end up somewhere in 1970).

hm seems like different behavior would be useful for native as a test env for mcu-code (where where rtc starts at 0 after reset) and native as a target-platform to run your code on.

Reply:
The current behavior of `periph_rtc` on `native` is to return the wall clock.
This PR only adds sub-second precision to it.

Reply:
I tried with

```patch
--- a/cpu/native/periph/rtc.c
+++ b/cpu/native/periph/rtc.c
@@ -155,7 +155,7 @@ int rtc_set_time(struct tm *ttime)
 
 int rtc_get_time_ms(struct tm *ttime, uint16_t *ms)
 {
-    struct timeval tv;
+    struct timespec tv;
 
     if (!_native_rtc_initialized) {
         warnx("rtc_get_time: not initialized");
@@ -167,11 +167,11 @@ int rtc_get_time_ms(struct tm *ttime, uint16_t *ms)
     }
 
     _native_syscall_enter();
-    gettimeofday(&tv, NULL);
+    clock_gettime(CLOCK_REALTIME, &tv);
     tv.tv_sec += _native_rtc_offset;
 
     if (ms) {
-        *ms = tv.tv_usec / US_PER_MS;
+        *ms = tv.tv_nsec / NS_PER_MS;
     }
 
     if (localtime_r(&tv.tv_sec, ttime) == NULL) {

``` 

and I'm not getting the real time:

```
RIOT RTC low-level driver test
This test will display 'Alarm!' every 2 seconds for 4 times
Native RTC initialized.
  Setting clock to   2020-02-28 23:59:57
Clock value is now   2020-02-28 23:59:58.000
  Setting alarm to   2020-02-28 23:59:60
   Alarm is set to   2020-02-28 23:59:60
  Alarm cleared at   2020-02-28 23:59:58.000
       No alarm at   2020-02-29 00:00:00.000
  Setting alarm to   2020-02-29 00:00:02

Alarm!
Alarm!
Alarm!
Alarm!
```

But I also really don't see the advantage of this.

Reply:
huh, nvm I also get those results with `gettimeofday()`

Reply:
Well if you insist - I really don't see the advantage of this.